### PR TITLE
refactor: remove unnecessary deps from 08-wasm by reusing top level simapp binary

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

The general idea:
* top level simapp should contain all our modules and support a binary for all e2e tests 
* adds wasm to top level simapp, along with build requirements in workflow + Dockerfile
* left 08-wasm Docker image for release building (instead of adding more conditionals into the workflows for the switch over point)
* e2e's run off CI should use the simd image, running against older branches requires using existing release tag, or building new image via manual wasm build
* left wasm e2e workflow to prevent wasm e2e test from being run on every pr

closes: #7231 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: simapp/app.go
Comment (colin-axner) [inline simapp/app.go:617]: copied this over, is this todo still relevant? I'm not sure I see how the light client module could be passed?

File: .github/workflows/build-simd-image-from-tag.yml
Comment (colin-axner) [inline .github/workflows/build-simd-image-from-tag.yml:1]: should I try to remove all the custom wasm workflows? I feel like now that we have a single place for a simapp, we should just be building a single simd. What do people think?

File: .github/workflows/docker.yml
Comment (colin-axner) [inline .github/workflows/docker.yml:1]: would be nice if someone could double check all the workflow changes, especially release.yml

File: .github/workflows/build-simd-image-from-tag.yml
Comment (colin-axner) [inline .github/workflows/build-simd-image-from-tag.yml:31]: I guess this is necessary to use the script to get the libwasm version and checksum

File: modules/light-clients/08-wasm/Dockerfile
Comment (colin-axner) [inline modules/light-clients/08-wasm/Dockerfile:1]: should I simply delete this file now?

File: Dockerfile
Comment (colin-axner) [inline Dockerfile:38]: It looks like `-linkmode=external -extldflags "-Wl,-z,muldefs -static"` is not usually included when we run `make build`. Should I try to include these changes there as well? Folks okay if I hardcode it for now?

Comment (colin-axner) [reply Dockerfile:38]: did a quick peek into changing this, but ran into a little difficulties, so I'll let someone be my guest if they want to adjust it

File: modules/light-clients/08-wasm/testing/simapp/simd/main.go
Comment (colin-axner) [reply modules/light-clients/08-wasm/testing/simapp/simd/main.go:16]: I like that direction. I can split this work up:
* add wasm into top level simapp
* switch over wasm workflows to main workflow (I think there's some consideration around image building off release branches)
* remove simd in wasm directory + Dockerfile (removes unnecessary deps)

File: modules/light-clients/08-wasm/go.mod
Comment (colin-axner) [inline modules/light-clients/08-wasm/go.mod:12]: damn, 7 dependencies :axe:

File: modules/light-clients/08-wasm/testing/simapp/simd/main.go
Comment (colin-axner) [reply modules/light-clients/08-wasm/testing/simapp/simd/main.go:16]: I went through the reference in the fork workflow and it looks like running wasm tests on forks is currently disabled

**Review by DimitrisJim — APPROVED** [review]
won't smash through, will wait for another review. looks good to me! :1st_place_medal:

**Comment by damiannolan** [comment]
Would probably recommend to redo this PR against a clean HEAD of main. 
I think that's an easier lift than handling all the conflicts here.

**Comment by DimitrisJim** [comment]
> Would probably recommend to redo this PR against a clean HEAD of main.

Indeed, will close and leave a comment on issue!

**Pull request closed without merging**
